astronomer/providers/snowflake/triggers/snowflake_sql_api_trigger.py

```diff
@@ -0,0 +1,89 @@
+import asyncio
+from datetime import timedelta
+from typing import Any, AsyncIterator, Dict, List, Tuple
+
+from airflow.triggers.base import BaseTrigger, TriggerEvent
+
+from astronomer.providers.snowflake.hooks.snowflake_sql_api import (
+    SnowflakeSqlApiHookAsync,
+)
+
+
+class SnowflakeSqlApiTrigger(BaseTrigger):
+    """
+    SnowflakeSqlApi Trigger inherits from the BaseTrigger,it is fired as
+    deferred class with params to run the task in trigger worker and
+    fetch the status for the query ids passed
+
+    :param task_id: Reference to task id of the Dag
+    :param poll_interval:  polling period in seconds to check for the status
+    :param query_ids: List of Query ids to run and poll for the status
+    :param snowflake_conn_id: Reference to Snowflake connection id
+    """
+
+    def __init__(
+        self,
+        task_id: str,
+        poll_interval: float,
+        query_ids: List[str],
+        snowflake_conn_id: str,
+        token_life_time: timedelta,
+        token_renewal_delta: timedelta,
+    ):
+        super().__init__()
+        self.task_id = task_id
+        self.poll_interval = poll_interval
+        self.query_ids = query_ids
+        self.snowflake_conn_id = snowflake_conn_id
+        self.token_life_time = token_life_time
+        self.token_renewal_delta = token_renewal_delta
+
+    def serialize(self) -> Tuple[str, Dict[str, Any]]:
+        """Serializes SnowflakeTrigger arguments and classpath."""
+        return (
+            "astronomer.providers.snowflake.triggers.snowflake_sql_api_trigger.SnowflakeSqlApiTrigger",
+            {
+                "task_id": self.task_id,
+                "poll_interval": self.poll_interval,
+                "query_ids": self.query_ids,
+                "snowflake_conn_id": self.snowflake_conn_id,
+                "token_life_time": self.token_life_time,
+                "token_renewal_delta": self.token_renewal_delta,
+            },
+        )
+
+    async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
+        """
+        Makes a GET API request to snowflake with query_id to get the status of the query
+        by get_sql_api_query_status async function
+        """
+        hook = SnowflakeSqlApiHookAsync(
+            self.snowflake_conn_id, self.token_life_time, self.token_renewal_delta
+        )
+        try:
+            statement_query_ids: List[str] = []
+            for query_id in self.query_ids:
+                while await self.is_still_running(query_id):
+                    await asyncio.sleep(1)
```

Comment:
why sleep for 1 sec? please use `self. poll_interval`

Reply:
Fixed